test/NuGet.Core.Tests/NuGet.Packaging.Test/ReferencesInNuspecMatchRefAssetsRuleTests.cs

```diff
@@ -0,0 +1,408 @@
+            var references = new Dictionary<string, IEnumerable<string>>()
+            {
+                { "MyCompany", new List<string>() { "MyLib.dll" } }
+            };
+            var files = new string[]
+            {
+                "ref/MyCompany/MyLib.dll",
+                "ref/MyCompany/MyHelpers.dll"
+            };
+
+            //Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingReferences = rule.Compare(references, files);
+
+            //Assert
+            Assert.Empty(missingReferences);
+        }
+
+        [Fact]
+        public void Compare_PackageWithMismatchedReferencesInMultipleSubFolders_ShouldHaveCorrectItems()
+        {
+            //Arrange
+            var references = new Dictionary<string, IEnumerable<string>>
+            {
+                { "net462", new List<string>() { "MyLib.dll", "MyHelpers.dll" } },
+                { "net472", new List<string>() { "MyLib.dll" } }
+            };
+            var files = new string[]
+            {
+                "ref/net462/MyLib.dll",
+                "ref/net472/MyLib.dll",
+                "ref/net472/MyHelpers.dll"
+            };
+
+            //Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingReferences = rule.Compare(references, files);
+
+            //Assert
+            Assert.Equal(missingReferences.Count(), 2);
+            var singleRefFilesMissing = missingReferences.Single(t => t.MissingFrom.Equals("ref"));
+            var singleReferencesMissing = missingReferences.Single(t => t.MissingFrom.Equals("nuspec"));
+            Assert.Equal("net462", singleRefFilesMissing.Tfm);
+            Assert.Equal(1, singleRefFilesMissing.MissingItems.Count());
+            Assert.Equal("net472", singleReferencesMissing.Tfm);
+            Assert.Equal(1, singleReferencesMissing.MissingItems.Count());
+            Assert.Equal("MyHelpers.dll", singleReferencesMissing.MissingItems[0]);
+        }
+
+        [Fact]
+        public void Compare_PackageHasReferencesWithNoRefFiles_ShouldContainCorrectFile()
+        {
+            //Arrange
+            var references = new Dictionary<string, IEnumerable<string>>
+            {
+                { "net462", new List<string>() { "MyLib.dll" } }
+            };
+            var files = Array.Empty<string>();
+
+            //Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingReferences = rule.Compare(references, files);
+
+            //Assert
+            Assert.Equal(missingReferences.Count(), 1);
+            var singleIssue = missingReferences.Single(t => t.Tfm.Equals("net462"));
+            Assert.Equal("net462", missingReferences.First().Tfm);
+            Assert.Equal(1, missingReferences.First().MissingItems.Count());
+            Assert.Equal("MyLib.dll", missingReferences.First().MissingItems[0]);
+            Assert.Equal("ref", missingReferences.First().MissingFrom);
```

Comment:
just FYI: while `missingReferences.First()` isn't very computationally expensive, given there are thousands of tests, little things can add up.  Don't be afraid to declare a variable.